src/shims/native_lib/trace/child.rs

```diff
@@ -130,6 +120,18 @@ impl Supervisor {
                 .unwrap();
             }
 
+            // We can't use IPC channels here to signal that FFI mode has ended,
+            // since they might allocate memory which could get us stuck in a SIGTRAP
+            // with no easy way out! While this could be worked around, it is much
```

Comment:
I don't understand this comment. What is the problem with allocating memory / SIGTRAP?

Reply:
Right now there's none (I guess it's a leftover comment I didn't properly update when I cut out the malloc tracing), but I implemented it this way from the start because of the way I intended to implement said malloc tracing bits. Shouldn't be of concern to this PR, though